packages/react-router/__tests__/path-matching-test.tsx

```diff
@@ -515,6 +581,170 @@ describe("path matching with optional dynamic segments", () => {
   });
 
   test("consecutive optional dynamic segments in nested routes", () => {
+    let manuallyExploded = [
```

Comment:
There's some ambiguity here - i.e., would the user put the value there first or not there first?  I chose to assume that since we want to match deeper paths, that we would assume you would put the param there first, followed by the version without the param.